### ISSUE TITLE
Generated ApplyConfiguration doesn't handle byte slices properly

### ISSUE THREAD

**Issue by dprotaso** [kubernetes/kubernetes#121030]
### What happened?

`[]byte` slices aren't handled correctly in the generated apply configuration code

https://github.com/kubernetes/kubernetes/blob/57144165f7afec3db71ca6f081ee8f4e4dfeab8a/staging/src/k8s.io/client-go/applyconfigurations/admissionregistration/v1/webhookclientconfig.go#L54-L59

### What did you expect to happen?

I expected the byte slice to be assigned to the property

eg.

```go
// WithCABundle adds the given value to the CABundle field in the declarative configuration
// and returns the receiver, so that objects can be build by chaining "With" function invocations.
// If called multiple times, values provided by each call will be appended to the CABundle field.
func (b *WebhookClientConfigApplyConfiguration) WithCABundle(value []byte) *WebhookClientConfigApplyConfiguration {
	b.CABundle = value
	return b
}
```

### How can we reproduce it (as minimally and precisely as possible)?

See code generator

### Kubernetes version

All versions

### Cloud provider

N/A

**Comment by aojea** [comment]
ping @apelisse

**Comment by apelisse** [comment]
Yeah, the algorithm is that we append to lists, but I agree it doesn't make sense for a []byte. Hopefully `[]byte` are always treated as atomic.

**Comment by seans3** [comment]
/triage accepted

@siyuanfoundation  No hurry, but do you have any information on the current status of this. We saw the PR to address this was closed. Thanks.

**Comment by dprotaso** [comment]
/lifecycle frozen

@siyuanfoundation @jpbetz is this still an issue?
